addons/l10n_fr_pos_cert/static/src/js/l10n_fr_pos_cert.js

```diff
@@ -0,0 +1,247 @@
+                            });
+                            certification_deferred.resolve(hash);
+                            return server_ids;
+                        }).fail(function (error, event){
+                            certification_deferred.reject();
+                            return server_ids;
+                        });
+                    }
+                    certification_deferred.resolve(false);
+                    return server_ids;
+                }
+                certification_deferred.reject();
+            }, function error() {
+                certification_deferred.reject();
+            });
+        },
+
+    });
+
+    /*************************************************************************
+        [if iface_print_via_proxy IS NOT enabled]
+
+        Extend ReceiptScreenWidget:
+            - overload show() function to wait for the order hash,
+              before printing the bill.
+     */
+    var ReceiptScreenWidgetShowParent = screens.ReceiptScreenWidget.prototype.show;
+    screens.ReceiptScreenWidget.include({
+
+        show: function(){
+            var self = this;
+            var setting = this.pos.config.l10n_fr_print_hash;
+            if (setting === 'no_print'){
+                // Direct Call
+                self.show_certification();
+            } else {
+                // Wait for Promise
+                certification_deferred.then(function success(order_id) {
+                    self.show_certification();
+                }, function error() {
+                    self.show_certification();
+                });
+            }
+        },
+
+        show_certification: function(){
+            // Display the bill for printing
+            ReceiptScreenWidgetShowParent.apply(this, []);
+            certification_deferred = null;
+        },
+    });
+
+    /*************************************************************************
+        [if iface_print_via_proxy IS enabled]
+
+        Extend module.ProxyDevice:
+            - overload print_receipt() function to wait for the order hash,
+              before printing the bill.
+     */
+    var ProxyDevicePrintReceiptParent = devices.ProxyDevice.prototype.print_receipt;
+    devices.ProxyDevice.include({
+
+        print_receipt: function(receipt){
+            var self = this;
+            var setting = this.pos.config.l10n_fr_print_hash;
+            if (receipt){
+                if (setting === 'no_print'){
+                    self.print_receipt_certification(receipt, false);
+                } else {
+                    certification_deferred.then(function success(hash) {
+                        self.print_receipt_certification(receipt, hash);
+                    }, function error() {
+                        self.print_receipt_certification(receipt, false);
+                    });
+                }
+            } else {
+                // Weird core feature, print_receipt is called regularly
```

Comment:
I think it is because, each time the printer get connected, it retries to print tickets that may be waiting (but @jorenvo can probably confirm this)

Reply:
Yes, what I don't exactly understand, is why sometimes, print_receipt is call without receipt arg. (receipt=False). but I wrote this comment 6 monthes ago, don't remember exactly the context. I can remove it if you want.